src/urljoin.php

```diff
@@ -18,13 +18,34 @@ function urljoin($base, $rel) {
 		return $base;
 	}
 
+	$is_valid_scheme_fn = function ($scheme) {
```

Comment:
Hm, even if this is necessary (and I don't think it is), it would be much better to do this with `preg_match`, e.g.:

```php
return preg_match('^[a-zA-Z0-9\-+.]*$');
```

However, see my other comment for what seems to be a better way to do this...